src/Files.App.Storage/NativeStorage/NativeFolder.cs

```diff
@@ -0,0 +1,204 @@
+            {
+                File.Delete(locatableFile.Path);
+            }
+            else if (item is ILocatableFolder locatableFolder)
+            {
+                Directory.Delete(locatableFolder.Path, true);
+            }
+            else
+                throw new ArgumentException($"Could not delete {item}.");
+
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc/>
+        public async Task<IStorable> CreateCopyOfAsync(IStorable itemToCopy, CreationCollisionOption collisionOption = default, CancellationToken cancellationToken = default)
+        {
+            var overwrite = collisionOption == CreationCollisionOption.ReplaceExisting;
+
+            if (itemToCopy is IFile sourceFile)
+            {
+                if (itemToCopy is ILocatableFile sourceLocatableFile)
+                {
+                    var newPath = System.IO.Path.Combine(Path, itemToCopy.Name);
+                    File.Copy(sourceLocatableFile.Path, newPath, overwrite);
+
+                    return new NativeFile(newPath);
+                }
+
+                var copiedFile = await CreateFileAsync(itemToCopy.Name, collisionOption, cancellationToken);
+                await sourceFile.CopyContentsToAsync(copiedFile, cancellationToken);
+
+                return copiedFile;
+            }
+            else if (itemToCopy is IFolder sourceFolder)
+            {
+                // TODO: Implement folder copy
+                throw new NotSupportedException();
+            }
+
+            throw new ArgumentException($"Could not copy type {itemToCopy.GetType()}");
+        }
+
+        /// <inheritdoc/>
+        public async Task<IStorable> MoveFromAsync(IStorable itemToMove, IModifiableFolder source, CreationCollisionOption collisionOption = default, CancellationToken cancellationToken = default)
+        {
+            var overwrite = collisionOption == CreationCollisionOption.ReplaceExisting;
+
+            if (itemToMove is IFile sourceFile)
+            {
+                if (itemToMove is ILocatableFile sourceLocatableFile)
+                {
+                    var newPath = System.IO.Path.Combine(Path, itemToMove.Name);
+                    File.Move(sourceLocatableFile.Path, newPath, overwrite);
+
+                    return new NativeFile(newPath);
+                }
+                else
+                {
+                    var copiedFile = await CreateFileAsync(itemToMove.Name, collisionOption, cancellationToken);
+                    await sourceFile.CopyContentsToAsync(copiedFile, cancellationToken);
+                    await source.DeleteAsync(itemToMove, true, cancellationToken);
+
+                    return copiedFile;
+                }
+            }
+            else if (itemToMove is IFolder sourceFolder)
+            {
+                throw new NotImplementedException();
+            }
+
+            throw new ArgumentException($"Could not move type {itemToMove.GetType()}");
+        }
```

Comment:
`MoveAsync` and `CreateCopy` are pretty close in logic. Maybe they could be using the same method taking a `Func` in its behavior or a flag?

Reply:
I think it's good time to revise on codestyle guidelines we have on the website. It's been quite some time since they have been last updated

Reply:
Func would be unclear and it's generally a not a good practice. If I tried really hard, I perhaps could get rid of some code duplication but then it'd get too cluttered